core/arch/arm/plat-hikey/main.c

```diff
@@ -93,3 +108,166 @@ void console_flush(void)
 {
 	pl011_flush(console_base());
 }
+
+static vaddr_t peripherals_base(paddr_t pa)
+{
+	void *va;
+
+	if (cpu_mmu_enabled()) {
+		va = phys_to_virt(pa, MEM_AREA_IO_NSEC);
+		return (vaddr_t)va;
+	}
+	return (vaddr_t)pa;
+}
+
+#ifdef CFG_SPI
+#ifdef CFG_SPI_TEST
+static void spi_test(void)
+{
+	struct pl061_data platform_pl061_data;
+	struct pl022_data platform_pl022_data;
+
+	vaddr_t gpio6_base = peripherals_base(GPIO6_BASE);
+	vaddr_t spi_base = peripherals_base(SPI_BASE);
+
+	uint8_t tx[3] = {1, 0x80, 0};
```

Comment:
Would it be possible to clarify what the values actually will do. I understand that it's the sequence to run som SPI test. I.e, maybe a description at the beginning of the function stating a bit more what is sitting on the other end when running this test would help.